python/cudf/cudf/pandas/module_accelerator.py

```diff
@@ -616,6 +623,19 @@ def install(
 def disable_module_accelerator() -> contextlib.ExitStack:
     """
     Temporarily disable any module acceleration.
+
+    This function only offers limited guarantees of thread safety.
+    Cases that will work:
+        - multiple threads are launched and each independently turns off acceleration
+        - a single thread turns off acceleration and then launches multiple threads
+          inside the context manager
+
+    Cases that trigger race conditions:
+        - a single thread launches multiple threads and then enters the context manager
+          while those threads are still running
+        - nested thread launching and acceleration disabling, i.e. if a thread launches
+          a that disables acceleration and then launches another thread, the innermost
+          thread will not have the accelerator disabled.
```

Comment:
```suggestion
        - nested thread launching and acceleration disabling, i.e. if a thread disables
          acceleration and then launches another thread, the innermost thread will not
          have the accelerator disabled.
```

Reply:
There's a typo in my statement that's probably confusing you, but your changes are actually not equivalent. The issue I'm trying to describe is when there are two levels of nesting, whereas your description changes there to only be one.
```suggestion
        - nested thread launching and acceleration disabling, i.e. if a thread launches
          a thread that disables acceleration and then launches another thread, the
          innermost thread will not have the accelerator disabled.
```

Reply:
Oh yeah you're right the typo confused me. Thanks for fixing that